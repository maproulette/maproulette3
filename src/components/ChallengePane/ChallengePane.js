import React, { Component } from 'react'
import classNames from 'classnames'
import MediaQuery from 'react-responsive'
import _isEqual from 'lodash/isEqual'
import _get from 'lodash/get'
import ChallengeFilterSubnav from './ChallengeFilterSubnav/ChallengeFilterSubnav'
import MapPane from '../EnhancedMap/MapPane/MapPane'
import Sidebar from '../Sidebar/Sidebar'
import ChallengeSearchMap from '../ChallengeSearchMap/ChallengeSearchMap'
import ChallengeBrowseMap from '../ChallengeBrowseMap/ChallengeBrowseMap'
import CongratulateModal from '../CongratulateModal/CongratulateModal'
import ChallengeResultList from './ChallengeResultList/ChallengeResultList'
import WithChallenges from '../HOCs/WithChallenges/WithChallenges'
import WithStartChallenge from '../HOCs/WithStartChallenge/WithStartChallenge'
import WithFilteredChallenges
       from '../HOCs/WithFilteredChallenges/WithFilteredChallenges'
import WithChallengeSearch from '../HOCs/WithSearch/WithChallengeSearch'
import WithSearchResults from '../HOCs/WithSearchResults/WithSearchResults'
import WithBrowsedChallenge from '../HOCs/WithBrowsedChallenge/WithBrowsedChallenge'
import WithClusteredTasks from '../HOCs/WithClusteredTasks/WithClusteredTasks'
import WithTaskMarkers from '../HOCs/WithTaskMarkers/WithTaskMarkers'
import WithMapBoundedTasks from '../HOCs/WithMapBoundedTasks/WithMapBoundedTasks'
import WithStatus from '../HOCs/WithStatus/WithStatus'
import './ChallengePane.scss'

// Setup child components with necessary HOCs
const ChallengeResults = WithStatus(ChallengeResultList)
const BrowseMap = WithTaskMarkers(ChallengeBrowseMap)
let SearchMap = null

// If the map-bounded task browsing feature is enabled, set up the ChallengeSearchMap
// to use it.
if (_get(process.env,
         'REACT_APP_FEATURE_BOUNDED_TASK_BROWSING') === 'enabled') {
  SearchMap = WithTaskMarkers(ChallengeSearchMap, 'mapBoundedTasks')
}
else {
  SearchMap = ChallengeSearchMap
}

/**
 * ChallengePane represents the top-level view when the user is browsing,
 * searching, and choosing a challenge to start working on. It includes
 * a ChallengeFilterSubnav that presents a subnav with filter and search
 * options, a ChallengeResultList (wrapped in a Sidebar) for presenting challenges
 * that match the current search and set of filters, and a ChallengeSearchMap for
 * finding challenges geographically.
 *
 * @see See [ChallengeFilterSubnav](#challengefiltersubnav)
 * @see See [ChallengeResultList](#challengeresultlist)
 * @see See [ChallengeSearchMap](#ChallengeSearchMap)
 *
 * @author [Neil Rotstan](https://github.com/nrotstan)
 */
export class ChallengePane extends Component {
  state = {
    sidebarMinimized: true,
  }

  toggleSidebarMinimized = () => {
    this.setState({sidebarMinimized: !this.state.sidebarMinimized})
  }

  shouldComponentUpdate(nextProps, nextState) {
    return !_isEqual(this.props, nextProps) || !_isEqual(this.state, nextState)
  }

  render() {
    const Map = this.props.browsedChallenge ? BrowseMap : SearchMap
    return (
      <span>
        {_get(this.props, 'history.location.state.congratulate', false) &&
         <CongratulateModal />
        }
        <ChallengeFilterSubnav {...this.props} />

        <div className="challenge-pane">
          <MediaQuery maxWidth={900}>
            <Sidebar className={classNames('inline full-screen-height with-shadow challenge-pane__results',
                                          {"is-minimized": this.state.sidebarMinimized})}
                    toggleMinimized={this.toggleSidebarMinimized}
                    isActive={true}>
              <ChallengeResults {...this.props} />
            </Sidebar>
          </MediaQuery>
          <MediaQuery minWidth={901}>
            <Sidebar className='inline full-screen-height with-shadow challenge-pane__results'
                     isActive={true}>
              <ChallengeResults {...this.props} />
            </Sidebar>
          </MediaQuery>

          <MapPane>
            <Map challenge={this.props.browsedChallenge}
                 onTaskClick={this.props.startChallengeWithTask}
                 {...this.props} />
          </MapPane>
        </div>
      </span>
    )
  }
}

export default
  WithChallenges(
    WithChallengeSearch(
      WithFilteredChallenges(
        WithSearchResults(
          WithMapBoundedTasks(
            WithClusteredTasks(
              WithStartChallenge(
                WithBrowsedChallenge(ChallengePane)
              )
            )
          ),
          'challenges',
          'challenges'
        )
      )
    )
  )
