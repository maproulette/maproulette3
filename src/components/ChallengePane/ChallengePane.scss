@import '../../variables.scss';

.challenge-pane {
  position: relative;
  display: flex;

  .sidebar {
    background-color: $grey-lightest-more;
  }
}
