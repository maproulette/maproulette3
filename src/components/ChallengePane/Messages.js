import { defineMessages } from 'react-intl'

/**
 * Internationalized messages for use with ChallengePane
 */
export default defineMessages({
  startChallengeLabel: {
    id: "ChallengePane.controls.startChallenge.label",
    defaultMessage: "Start Challenge"
  }
})
