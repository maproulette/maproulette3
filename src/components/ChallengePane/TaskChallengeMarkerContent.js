import React, { Component } from 'react'
import _get from 'lodash/get'
import WithLoadedTask from '../HOCs/WithLoadedTask/WithLoadedTask'
import messages from './Messages'

/**
 * The content to show in the popup when a task marker is clicked.
 */
class TaskChallengeMarkerContent extends Component {
  render() {
    const markerData = this.props.marker
    let challengeId = markerData.options.parentId ||
                     _get(this.props.task, 'parent.id')
    if (!challengeId &&
        _get(markerData.options, 'challengeIds.length', 0) === 1) {
      challengeId = markerData.options.challengeIds[0]
    }

    return (
      <div className="marker-popup-content">
        <h3>
          {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
          <a onClick={() => this.props.history.push(
            `/browse/challenges/${challengeId}`
          )}>
            {markerData.options.parentName || _get(this.props.task, 'parent.name')}
          </a>
        </h3>

        <div className="marker-popup-content__links">
          <div>
            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
            <a onClick={() => {
              this.props.startChallengeWithTask(
                challengeId,
                false,
                markerData.options.taskId)
            }}>
              {this.props.intl.formatMessage(messages.startChallengeLabel)}
            </a>
          </div>
        </div>
      </div>
    )
  }
}

export default WithLoadedTask(TaskChallengeMarkerContent)
