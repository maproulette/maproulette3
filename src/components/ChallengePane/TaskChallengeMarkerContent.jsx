import { Component } from "react";
import WithLoadedTask from "../HOCs/WithLoadedTask/WithLoadedTask";
import messages from "./Messages";

/**
 * The content to show in the popup when a task marker is clicked.
 */
class TaskChallengeMarkerContent extends Component {
  render() {
    const markerData = this.props.marker;
    let challengeId = markerData.options.parentId || this.props.task?.parent?.id;
    if (!challengeId && (markerData.options?.challengeIds?.length ?? 0) === 1) {
      challengeId = markerData.options.challengeIds[0];
    }

    return (
      <div>
        <h3>
          <div className="mr-text-left mr-text-grey">
            <h4>{this.props.intl.formatMessage(messages.taskInfoLabel)}</h4>
            <div className="mr-text-sm">
              {this.props.intl.formatMessage(messages.taskNameLabel)}{" "}
              {markerData.options.name || this.props.task?.name}
              <br />
              {this.props.intl.formatMessage(messages.taskIdLabel)}{" "}
              <a
                onClick={() => {
                  this.props.startChallengeWithTask(challengeId, false, markerData.options.taskId);
                }}
              >
                {markerData.options.id || this.props.task?.id}
              </a>
              <br />
              {this.props.intl.formatMessage(messages.challengeNameLabel)}{" "}
              {markerData.options.parentName || this.props.task?.parent?.name}
              <br />
              {this.props.intl.formatMessage(messages.challengeIdLabel)}{" "}
              <a onClick={() => this.props.history.push(`/browse/challenges/${challengeId}`)}>
                {markerData.options.parentId || this.props.task?.parent?.id}
              </a>
            </div>
          </div>
        </h3>

        <div className="marker-popup-content__links">
          <div>
            <a
              onClick={() => {
                this.props.startChallengeWithTask(challengeId, false, markerData.options.taskId);
              }}
            >
              {this.props.intl.formatMessage(messages.startTaskLabel)}{" "}
              {markerData.options.id || this.props.task?.id}
            </a>
          </div>
        </div>
      </div>
    );
  }
}

export default WithLoadedTask(TaskChallengeMarkerContent);
