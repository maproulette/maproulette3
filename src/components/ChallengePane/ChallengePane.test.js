import React from 'react'
import ChallengePane from './ChallengePane'
import { ChallengeDifficulty }
       from '../../services/Challenge/ChallengeDifficulty/ChallengeDifficulty'

let basicProps = null

beforeEach(() => {
  basicProps = {
    user: {
      id: 11,
      savedChallenges: [],
    },
    startChallenge: jest.fn(),
    saveChallenge: jest.fn(),
    unsaveChallenge: jest.fn(),
    intl: {formatMessage: jest.fn()},
  }
})

test("renders with props as expected", () => {
  const wrapper = shallow(
    <ChallengePane {...basicProps} />
  )

  expect(wrapper.find('.challenge-pane').exists()).toBe(true)
  expect(wrapper).toMatchSnapshot()
})
