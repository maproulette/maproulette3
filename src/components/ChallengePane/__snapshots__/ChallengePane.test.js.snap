// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChallengePane
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
    saveChallenge={[MockFunction]}
    startChallenge={[MockFunction]}
    unsaveChallenge={[MockFunction]}
    user={
      Object {
        "id": 11,
        "savedChallenges": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <Component
          intl={
            Object {
              "formatMessage": [MockFunction],
            }
          }
          saveChallenge={[MockFunction]}
          startChallenge={[MockFunction]}
          unsaveChallenge={[MockFunction]}
          user={
            Object {
              "id": 11,
              "savedChallenges": Array [],
            }
          }
        />,
        <div
          className="challenge-pane"
        >
          <MediaQuery
            maxWidth={900}
            values={Object {}}
          >
            <Sidebar
              className="inline full-screen-height with-shadow challenge-pane__results is-minimized"
              isActive={true}
              toggleMinimized={[Function]}
            >
              <Connect(Connect(Component))
                intl={
                  Object {
                    "formatMessage": [MockFunction],
                  }
                }
                saveChallenge={[MockFunction]}
                startChallenge={[MockFunction]}
                unsaveChallenge={[MockFunction]}
                user={
                  Object {
                    "id": 11,
                    "savedChallenges": Array [],
                  }
                }
              />
            </Sidebar>
          </MediaQuery>
          <MediaQuery
            minWidth={901}
            values={Object {}}
          >
            <Sidebar
              className="inline full-screen-height with-shadow challenge-pane__results"
              isActive={true}
            >
              <Connect(Connect(Component))
                intl={
                  Object {
                    "formatMessage": [MockFunction],
                  }
                }
                saveChallenge={[MockFunction]}
                startChallenge={[MockFunction]}
                unsaveChallenge={[MockFunction]}
                user={
                  Object {
                    "id": 11,
                    "savedChallenges": Array [],
                  }
                }
              />
            </Sidebar>
          </MediaQuery>
          <Connect(MapPane)>
            <Connect(Connect(Connect(Component)))
              intl={
                Object {
                  "formatMessage": [MockFunction],
                }
              }
              saveChallenge={[MockFunction]}
              startChallenge={[MockFunction]}
              unsaveChallenge={[MockFunction]}
              user={
                Object {
                  "id": 11,
                  "savedChallenges": Array [],
                }
              }
            />
          </Connect(MapPane)>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "intl": Object {
            "formatMessage": [MockFunction],
          },
          "saveChallenge": [MockFunction],
          "startChallenge": [MockFunction],
          "unsaveChallenge": [MockFunction],
          "user": Object {
            "id": 11,
            "savedChallenges": Array [],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <MediaQuery
              maxWidth={900}
              values={Object {}}
            >
              <Sidebar
                className="inline full-screen-height with-shadow challenge-pane__results is-minimized"
                isActive={true}
                toggleMinimized={[Function]}
              >
                <Connect(Connect(Component))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />
              </Sidebar>
            </MediaQuery>,
            <MediaQuery
              minWidth={901}
              values={Object {}}
            >
              <Sidebar
                className="inline full-screen-height with-shadow challenge-pane__results"
                isActive={true}
              >
                <Connect(Connect(Component))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />
              </Sidebar>
            </MediaQuery>,
            <Connect(MapPane)>
              <Connect(Connect(Connect(Component)))
                intl={
                  Object {
                    "formatMessage": [MockFunction],
                  }
                }
                saveChallenge={[MockFunction]}
                startChallenge={[MockFunction]}
                unsaveChallenge={[MockFunction]}
                user={
                  Object {
                    "id": 11,
                    "savedChallenges": Array [],
                  }
                }
              />
            </Connect(MapPane)>,
          ],
          "className": "challenge-pane",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Sidebar
                className="inline full-screen-height with-shadow challenge-pane__results is-minimized"
                isActive={true}
                toggleMinimized={[Function]}
              >
                <Connect(Connect(Component))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />
              </Sidebar>,
              "maxWidth": 900,
              "values": Object {},
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(Connect(Component))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />,
                "className": "inline full-screen-height with-shadow challenge-pane__results is-minimized",
                "isActive": true,
                "toggleMinimized": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "intl": Object {
                    "formatMessage": [MockFunction],
                  },
                  "saveChallenge": [MockFunction],
                  "startChallenge": [MockFunction],
                  "unsaveChallenge": [MockFunction],
                  "user": Object {
                    "id": 11,
                    "savedChallenges": Array [],
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Sidebar
                className="inline full-screen-height with-shadow challenge-pane__results"
                isActive={true}
              >
                <Connect(Connect(Component))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />
              </Sidebar>,
              "minWidth": 901,
              "values": Object {},
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(Connect(Component))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />,
                "className": "inline full-screen-height with-shadow challenge-pane__results",
                "isActive": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "intl": Object {
                    "formatMessage": [MockFunction],
                  },
                  "saveChallenge": [MockFunction],
                  "startChallenge": [MockFunction],
                  "unsaveChallenge": [MockFunction],
                  "user": Object {
                    "id": 11,
                    "savedChallenges": Array [],
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(Connect(Connect(Component)))
                intl={
                  Object {
                    "formatMessage": [MockFunction],
                  }
                }
                saveChallenge={[MockFunction]}
                startChallenge={[MockFunction]}
                unsaveChallenge={[MockFunction]}
                user={
                  Object {
                    "id": 11,
                    "savedChallenges": Array [],
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "challenge": undefined,
                "intl": Object {
                  "formatMessage": [MockFunction],
                },
                "onTaskClick": undefined,
                "saveChallenge": [MockFunction],
                "startChallenge": [MockFunction],
                "unsaveChallenge": [MockFunction],
                "user": Object {
                  "id": 11,
                  "savedChallenges": Array [],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <Component
            intl={
              Object {
                "formatMessage": [MockFunction],
              }
            }
            saveChallenge={[MockFunction]}
            startChallenge={[MockFunction]}
            unsaveChallenge={[MockFunction]}
            user={
              Object {
                "id": 11,
                "savedChallenges": Array [],
              }
            }
          />,
          <div
            className="challenge-pane"
          >
            <MediaQuery
              maxWidth={900}
              values={Object {}}
            >
              <Sidebar
                className="inline full-screen-height with-shadow challenge-pane__results is-minimized"
                isActive={true}
                toggleMinimized={[Function]}
              >
                <Connect(Connect(Component))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />
              </Sidebar>
            </MediaQuery>
            <MediaQuery
              minWidth={901}
              values={Object {}}
            >
              <Sidebar
                className="inline full-screen-height with-shadow challenge-pane__results"
                isActive={true}
              >
                <Connect(Connect(Component))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />
              </Sidebar>
            </MediaQuery>
            <Connect(MapPane)>
              <Connect(Connect(Connect(Component)))
                intl={
                  Object {
                    "formatMessage": [MockFunction],
                  }
                }
                saveChallenge={[MockFunction]}
                startChallenge={[MockFunction]}
                unsaveChallenge={[MockFunction]}
                user={
                  Object {
                    "id": 11,
                    "savedChallenges": Array [],
                  }
                }
              />
            </Connect(MapPane)>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "intl": Object {
              "formatMessage": [MockFunction],
            },
            "saveChallenge": [MockFunction],
            "startChallenge": [MockFunction],
            "unsaveChallenge": [MockFunction],
            "user": Object {
              "id": 11,
              "savedChallenges": Array [],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <MediaQuery
                maxWidth={900}
                values={Object {}}
              >
                <Sidebar
                  className="inline full-screen-height with-shadow challenge-pane__results is-minimized"
                  isActive={true}
                  toggleMinimized={[Function]}
                >
                  <Connect(Connect(Component))
                    intl={
                      Object {
                        "formatMessage": [MockFunction],
                      }
                    }
                    saveChallenge={[MockFunction]}
                    startChallenge={[MockFunction]}
                    unsaveChallenge={[MockFunction]}
                    user={
                      Object {
                        "id": 11,
                        "savedChallenges": Array [],
                      }
                    }
                  />
                </Sidebar>
              </MediaQuery>,
              <MediaQuery
                minWidth={901}
                values={Object {}}
              >
                <Sidebar
                  className="inline full-screen-height with-shadow challenge-pane__results"
                  isActive={true}
                >
                  <Connect(Connect(Component))
                    intl={
                      Object {
                        "formatMessage": [MockFunction],
                      }
                    }
                    saveChallenge={[MockFunction]}
                    startChallenge={[MockFunction]}
                    unsaveChallenge={[MockFunction]}
                    user={
                      Object {
                        "id": 11,
                        "savedChallenges": Array [],
                      }
                    }
                  />
                </Sidebar>
              </MediaQuery>,
              <Connect(MapPane)>
                <Connect(Connect(Connect(Component)))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />
              </Connect(MapPane)>,
            ],
            "className": "challenge-pane",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Sidebar
                  className="inline full-screen-height with-shadow challenge-pane__results is-minimized"
                  isActive={true}
                  toggleMinimized={[Function]}
                >
                  <Connect(Connect(Component))
                    intl={
                      Object {
                        "formatMessage": [MockFunction],
                      }
                    }
                    saveChallenge={[MockFunction]}
                    startChallenge={[MockFunction]}
                    unsaveChallenge={[MockFunction]}
                    user={
                      Object {
                        "id": 11,
                        "savedChallenges": Array [],
                      }
                    }
                  />
                </Sidebar>,
                "maxWidth": 900,
                "values": Object {},
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Connect(Connect(Component))
                    intl={
                      Object {
                        "formatMessage": [MockFunction],
                      }
                    }
                    saveChallenge={[MockFunction]}
                    startChallenge={[MockFunction]}
                    unsaveChallenge={[MockFunction]}
                    user={
                      Object {
                        "id": 11,
                        "savedChallenges": Array [],
                      }
                    }
                  />,
                  "className": "inline full-screen-height with-shadow challenge-pane__results is-minimized",
                  "isActive": true,
                  "toggleMinimized": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "intl": Object {
                      "formatMessage": [MockFunction],
                    },
                    "saveChallenge": [MockFunction],
                    "startChallenge": [MockFunction],
                    "unsaveChallenge": [MockFunction],
                    "user": Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Sidebar
                  className="inline full-screen-height with-shadow challenge-pane__results"
                  isActive={true}
                >
                  <Connect(Connect(Component))
                    intl={
                      Object {
                        "formatMessage": [MockFunction],
                      }
                    }
                    saveChallenge={[MockFunction]}
                    startChallenge={[MockFunction]}
                    unsaveChallenge={[MockFunction]}
                    user={
                      Object {
                        "id": 11,
                        "savedChallenges": Array [],
                      }
                    }
                  />
                </Sidebar>,
                "minWidth": 901,
                "values": Object {},
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Connect(Connect(Component))
                    intl={
                      Object {
                        "formatMessage": [MockFunction],
                      }
                    }
                    saveChallenge={[MockFunction]}
                    startChallenge={[MockFunction]}
                    unsaveChallenge={[MockFunction]}
                    user={
                      Object {
                        "id": 11,
                        "savedChallenges": Array [],
                      }
                    }
                  />,
                  "className": "inline full-screen-height with-shadow challenge-pane__results",
                  "isActive": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "intl": Object {
                      "formatMessage": [MockFunction],
                    },
                    "saveChallenge": [MockFunction],
                    "startChallenge": [MockFunction],
                    "unsaveChallenge": [MockFunction],
                    "user": Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(Connect(Connect(Component)))
                  intl={
                    Object {
                      "formatMessage": [MockFunction],
                    }
                  }
                  saveChallenge={[MockFunction]}
                  startChallenge={[MockFunction]}
                  unsaveChallenge={[MockFunction]}
                  user={
                    Object {
                      "id": 11,
                      "savedChallenges": Array [],
                    }
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "challenge": undefined,
                  "intl": Object {
                    "formatMessage": [MockFunction],
                  },
                  "onTaskClick": undefined,
                  "saveChallenge": [MockFunction],
                  "startChallenge": [MockFunction],
                  "unsaveChallenge": [MockFunction],
                  "user": Object {
                    "id": 11,
                    "savedChallenges": Array [],
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
